### PR TITLE
"TMI-rebased" - ping-pong

### PR DESCRIPTION
**Pull request by jMyles**

### CODE REVIEW

**Comment by KPrasch** [comment]
@jMyles - Looks like this might have the wrong base?

**Comment by jMyles** [comment]
Yeah - as I said, I'm updating TI so that this can be a PR on top of it.  IE, I'm sliding `4795a32ba` (with `39b5101f1` cherry-picked over it to make `65b3666d6`) underneath.  Stand by.

**Comment by jMyles** [comment]
@KPrasch: History is now fixed up.

File: tests/acceptance/cli/ursula/test_ursula_command.py [hunk 1]
```diff
@@ -39,7 +39,8 @@ def capture_output():
 @pytest.fixture(scope='module')
 def ursula(federated_ursulas):
     ursula = federated_ursulas.pop()
-    return ursula
+    yield ursula
+    ursula.disenchant()
```
Comment (derekpierre) [inline tests/acceptance/cli/ursula/test_ursula_command.py:43]: Should we do `ursula.stop()` here instead?

File: nucypher/characters/lawful.py [hunk 2]
```diff
@@ -1255,15 +1276,27 @@ def run(self,
...
-        self._availability_tracker.stop()
-        self.stop_learning_loop()
-        if not self.federated_only:
-            self.work_tracker.stop()
-        if self._arrangement_pruning_task.running:
-            self._arrangement_pruning_task.stop()
+        # Handles the shutdown of a partially initialized character.
+        with contextlib.suppress(AttributeError):  # TODO: Is this acceptable here, what are alternatives?
+            self._availability_tracker.stop()
+            self.stop_learning_loop()
```
Comment (derekpierre) [inline nucypher/characters/lawful.py:1295]: Should `self.stop_learning_loop()` instead be `super().disenchant()` which runs `disenchant()` from `Character`? Ultimately both do the same thing of stopping the learning loop by calling `Learner.stop_learning_loop()` but fulfils the subclass usage...

File: tests/acceptance/cli/ursula/test_ursula_command.py (hunk 1 above)
Comment (KPrasch) [reply tests/acceptance/cli/ursula/test_ursula_command.py:43]: Hmm - Somewhat of an open question re: combining `stop` and `disenchant` ( @jMyles )

File: nucypher/characters/lawful.py (hunk 2 above)
Comment (KPrasch) [reply nucypher/characters/lawful.py:1295]: Hmm - yes this is worth consideration -- using some common shutdown logic methods.

File: tests/acceptance/network/test_network_actors.py [hunk 3]
```diff
@@ -162,6 +162,7 @@ class FakeArrangement:
                                                    ursula=vladimir)
 
 
+@pytest.mark.skip('Needs to be restored -- no way to access treasure maps from stranger ursulas here.')
```
[the remark this answers is not in the document]
Comment (KPrasch) [reply tests/acceptance/network/test_network_actors.py:165]: Perhaps one issue to encapsulate all of the freshly skipped tests in #1741

**Comment by KPrasch** [comment]
Thanks for the reviews - let's ensure they are included in further consideration around #1741 :+1:

**Pull request merged**